File: sampleCode/CSharp/ConsoleApp/Endpoints/Events/IndustryEmploymentEvent.cs
namespace ConsoleApp.Endpoints.Events;

public record class IndustryEmploymentEvent : Event
{
    public double? Output { get; set; }
    public required double? Employment { get; set; }
    public double? EmployeeCompensation { get; set; }
    public double? ProprietorIncome { get; set; }
    public int IndustryCode { get; set; }

    public override string ImpactEventType => "IndustryEmployment";
}

File: sampleCode/R/Endpoints/Events/IndustryEmploymentEvent.R
source(auth_env$Event)

setClass("IndustryEmploymentEvent",
         contains = "Event",
         slots = list(
           Output = "numeric",
           Employment = "numeric",
           EmployeeCompensation = "numeric",
           ProprietorIncome = "numeric",
           IndustryCode = "integer",
           ImpactEventType = "character"
         ),
         prototype = list(
           ImpactEventType = "IndustryEmployment" # Assigning default values
         ))